src/openms/include/OpenMS/FILTERING/DATAREDUCTION/MassTraceDetection.h

```diff
@@ -97,27 +103,93 @@ namespace OpenMS
 
         /** @name Private methods and members
         */
+
+       
     protected:
         void updateMembers_() override;
 
     private:
 
         struct Apex
         {
-          Apex(double intensity, Size scan_idx, Size peak_idx);
-          double intensity;
-          Size scan_idx;
-          Size peak_idx;
+          Apex(PeakMap& map, const Size scan_idx, const Size peak_idx);
+          // Default constructors
+          Apex() = default;
+          Apex(const Apex& other) = default;
+          Apex(Apex&& other) = default;
+
+          // Move assignment operator
+          Apex& operator=(Apex&& other) 
+          {
+            if (this != &other) {
+              map_ = other.map_;
+              scan_idx_ = other.scan_idx_;
+              peak_idx_ = other.peak_idx_;
+            }
+            return *this;
+          }
+
+          PeakMap& map_;
```

Comment:
see https://en.cppreference.com/w/cpp/utility/functional/reference_wrapper

```suggestion
          std::reference_wrapper<PeakMap> map_;
```